localstack/services/awslambda/invocation/docker_runtime_executor.py

```diff
@@ -208,6 +211,9 @@ def _build_executor_endpoint(self, service_endpoint: ServiceEndpoint) -> Executo
             self.id,
         )
         executor_endpoint = ExecutorEndpoint(self.id, service_endpoint=service_endpoint)
+        if config.LAMBDA_ASF_DEV_PORT_EXPOSE:
```

Comment:
Can we merge that whole if block with the first one in start? It seems unnecessary that this happens at object creation time. We set the executor endpoint IP in the not-port-expose path in start() as well.
Especially get_free_tcp_port can take some time, so I would rather have it in the start method.

Reply:
Yes, that makes a lot of sense and reduces clutter because the constructor can stay generic this way